### ISSUE TITLE
5번 과제에 대한 자세한 설명 부탁드립니다.

### ISSUE THREAD

**Issue by dqdcisid** [wonnybunny/bookish-computing-machine#2]
이번 건 pdf 만 봐서는 잘 이해가 안 가네요. 간단한 text substitution 문제인가요? 보통 저런 문제 풀때 awk, perl 이랑 정규식 regular expression 이용할텐데.. C 를 사용해야 하나요?

**Comment by wonnybunny** [comment]
네, 간단한 substitution 문제라고 하셨습니다. binary search tree 를 이용해서 바꾸기만 하면 되는 과제입니다. 오늘 내일 중으로 교수님께 다시 한 번 여쭤보도록 하겠습니다만 아마 제가 말씀 드린게 다 일겁니다.

**Comment by wonnybunny** [comment]
방금 교수님으로부터 답을 받았는데, map file을 binary search tree를 이용해야 하는 점만 주의하면 된다고 하시네요.

**Comment by wonnybunny** [comment]
뉴비님 혹시 과제가 어느정도 진전 되었나요?

**Comment by dqdcisid** [comment]
근 일주일간 binary search tree 를 공부하면서 이리저리 구상해봤는데 당최 bst 를 map 파일에 이용하라는 의미를 모르겟읍니다.. 제 능력 밖의 일인 듯 하네요.. 유감이고 죄송합니다.
bst 를 사용해 정렬해서 최적화하라는 건지? 그렇다면 map file 을 정렬할 게 아니고 input text 자체를 정렬해야 하고 map file 정렬은 전혀 의미 없을텐데 .. ㅠㅜ

**Comment by dqdcisid** [comment]
혹시 교수님이나 프갤고수분들께 여쭤보시고 추가적인 힌트 얻으셨다면 추가 부탁드립니다. 6월 10일 시험이라 많이는 못 도와드리지만 힘 닿는 데 까지 노력해보겟읍니다.

**Comment by wonnybunny** [comment]
네, 뉴비님 다시 한 번 설명해 드릴게요. map file을
hello: hi
current: previous
이런식으로 작성을 해놓으면
한줄 한줄 씩 받아들여서 binary search tree를 구성하는 하나의 node(?)라고 부르는게 맞는지 모르겠는데, 아무튼 bst를 구성하는 공간에 저장을 시키는거에요. 한마디로 map file을 bst로 다시 한 번 구현하는거죠. 그 다음에 구현된 bst와 입력된 input txt와 비교하면서 해당하는 단어를 swap 해주는 과제입니다.
이해에 도움이 좀 되셨기를 바라며, 더 궁금한 점이 있으시면 달아주세요.

**Comment by wonnybunny** [comment]
뉴비님 과제 어느정도 되었나요?
